base/test/h264decoderNvCodec_tests.cpp

```diff
@@ -0,0 +1,172 @@
+#include "stdafx.h"
+#include <boost/test/unit_test.hpp>
+#include "FileReaderModule.h"
+#include "FileWriterModule.h"
+#include "Logger.h"
+#include "CudaMemCopy.h"
+#include "H264Decoder.h"
+#include "test_utils.h"
+#include "nv_test_utils.h"
+#include "PipeLine.h"
+#include "ExternalSinkModule.h"
+#include "H264Metadata.h"
+#include "H264DecoderNvCodecHelper.h"
+#include "Mp4ReaderSource.h"
+#include "Mp4VideoMetadata.h"
+#include "StatSink.h"
+
+BOOST_AUTO_TEST_SUITE(h264decodernvcodec_tests)
+
+BOOST_AUTO_TEST_CASE(h264_to_yuv420)
+{
+	Logger::setLogLevel("info");
+
+	// metadata is known
+	auto props = FileReaderModuleProps("./data/h264_data/FVDO_Freeway_4cif_???.H264", 0, -1);
+	props.readLoop = false;
+	auto fileReader = boost::shared_ptr<FileReaderModule>(new FileReaderModule(props));
+
+	auto h264ImageMetadata = framemetadata_sp(new H264Metadata(0, 0));
+
+	auto rawImagePin = fileReader->addOutputPin(h264ImageMetadata);
+
+	auto Decoder = boost::shared_ptr<Module>(new H264Decoder(H264DecoderProps()));
+	fileReader->setNext(Decoder);
+
+	auto fileWriter = boost::shared_ptr<Module>(new FileWriterModule(FileWriterModuleProps("./data/testOutput/yuv420Frames/Yuv420_704x576????.raw")));
+	Decoder->setNext(fileWriter);
+	fileReader->play(true);
```

Comment:
what is the significance of play(true) ?

Reply:
It basically increments the fileName index  and we can read all the files present in that dir.  play(true) sets the variable mPlay to true in fileNameStrategy ,which is used as a check to increment the index.

Reply:
In most of the tests we are trying to read only one single file, it is only required in case where we want to read whole stream of file in particular folder.